oak-run/src/main/js/oak-mongo.js

```diff
@@ -573,6 +595,201 @@ var oak = (function(global){
+            var isBson = (thetype == "[object BSON]");
+            if (!isBson) {
+                //print("   (not a bson, skipping " + k + ")");
+                continue;
+            }
+            var subdocsize = Object.bsonsize(subdoc);
+            if (subdocsize <= sizeLargerThan) {
+                //print("   (too small to report " + k + ")");
+                continue;
+            }
+            print(" - property " + k + " size : " + subdocsize);
+        }
+    }
+
+    /**
+     * Prtints the count of property revisions by clusterId
+     * The output is using the pseudo-revision format used elsewhere already
+     * where timestamp and count are 0 - mainly to point out the clusterId it belongs to
+     * This is useful for large documents to quickly see which clusterId was the
+     * most frequent writer.
+     *
+     * @memberof oak
+     * @method propertyClusterIdCounts
+     * @param {string} path the path of a document
+     */
+    api.propertyClusterIdCounts = function(path) {
+        if (path === undefined) {
+            print("No path specified");
+            return;
+        }
+        print("loading document at " + path);
+        var doc = this.findOne(path);
+        if (!doc) {
+            print("No document for path: " + path);
+            return;
+        }
+        var k;
+        var stats = {};
+        for (k in doc) {
+            if (k == "_id") {
+                continue;
+            }
+            var subdoc = doc[k];
+            var r;
+            var clusterIds = undefined;
+            for (r in subdoc) {
+                if (!subdoc.hasOwnProperty(r)) {
+                    continue;
+                }
+                var v = subdoc[r];
+                var clusterId = "r0-0-" + new Revision(r).getClusterId();
+                if (clusterIds === undefined) {
+                    clusterIds = {};
+                }
+                var existing = clusterIds[clusterId];
+                if (existing === undefined) {
+                    existing = 1;
+                } else {
+                    existing++;
+                }
+                clusterIds[clusterId] = existing;
+            }
+            if (clusterIds !== undefined) {
+                stats[k] = clusterIds;
+            }
+        }
+        // pretty format the output using stringify
+        print(JSON.stringify(stats, null, 8));
+    }
+
+    /**
+     * Prtints commit value of all branch commits.
+     *
+     * @memberof oak
+     * @method branchCommitValues
+     * @param {string} path the path of a document
+     */
+    api.branchCommitValues = function(path) {
+        if (path === undefined) {
+            print("No path specified");
+            return;
+        }
+        print("loading document at " + path);
+        var doc = this.findOne(path);
+        if (!doc) {
+            print("No document for path: " + path);
+            return;
+        }
+        var num = 0;
+        var r;
```

Comment:
I'm assuming "r" here is "revision": renaming to "rev" or adding a comment would make this a bit more clear.